src/cmd/linuxkit/run_virtualizationframework_darwin.go

```diff
@@ -0,0 +1,324 @@
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/Code-Hex/vz"
+	"github.com/pkg/term/termios"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	virtualizationNetworkingNone         string = "none"
+	virtualizationNetworkingDockerForMac        = "docker-for-mac"
+	virtualizationNetworkingVPNKit              = "vpnkit"
+	virtualizationNetworkingVMNet               = "vmnet"
+	virtualizationNetworkingDefault             = virtualizationNetworkingVMNet
+	virtualizationFrameworkConsole              = "console=hvc0"
+)
+
+// Process the run arguments and execute run
+func runVirtualizationFramework(args []string) {
+	flags := flag.NewFlagSet("virtualization", flag.ExitOnError)
+	invoked := filepath.Base(os.Args[0])
+	flags.Usage = func() {
+		fmt.Printf("USAGE: %s run virtualization [options] prefix\n\n", invoked)
+		fmt.Printf("'prefix' specifies the path to the VM image.\n")
+		fmt.Printf("\n")
+		fmt.Printf("Options:\n")
+		flags.PrintDefaults()
+	}
+	cpus := flags.Uint("cpus", 1, "Number of CPUs")
+	mem := flags.Uint64("mem", 1024, "Amount of memory in MB")
+	memBytes := *mem * 1024 * 1024
+	var disks Disks
+	flags.Var(&disks, "disk", "Disk config. [file=]path[,size=1G]")
+	data := flags.String("data", "", "String of metadata to pass to VM; error to specify both -data and -data-file")
+	dataPath := flags.String("data-file", "", "Path to file containing metadata to pass to VM; error to specify both -data and -data-file")
+
+	if *data != "" && *dataPath != "" {
+		log.Fatal("Cannot specify both -data and -data-file")
+	}
+
+	state := flags.String("state", "", "Path to directory to keep VM state in")
+	networking := flags.String("networking", virtualizationNetworkingDefault, "Networking mode. Valid options are 'default', 'vmnet' and 'none'. 'vmnet' uses the Apple vmnet framework. 'none' disables networking.`")
+
+	kernelBoot := flags.Bool("kernel", false, "Boot image is kernel+initrd+cmdline 'path'-kernel/-initrd/-cmdline")
+
+	if err := flags.Parse(args); err != nil {
+		log.Fatal("Unable to parse args")
+	}
+	remArgs := flags.Args()
+	if len(remArgs) == 0 {
+		fmt.Println("Please specify the prefix to the image to boot")
+		flags.Usage()
+		os.Exit(1)
+	}
+	path := remArgs[0]
+	prefix := path
+
+	_, err := os.Stat(path + "-kernel")
+	statKernel := err == nil
+
+	var isoPaths []string
+
+	// Default to kernel+initrd
+	if !statKernel {
+		log.Fatalf("Cannot find kernel file: %s", path+"-kernel")
+	}
+	_, err = os.Stat(path + "-initrd.img")
+	statInitrd := err == nil
+	if !statInitrd {
+		log.Fatalf("Cannot find initrd file (%s): %v", path+"-initrd.img", err)
+	}
+	*kernelBoot = true
+
+	metadataPaths, err := CreateMetadataISO(*state, *data, *dataPath)
+	if err != nil {
+		log.Fatalf("%v", err)
+	}
+	isoPaths = append(isoPaths, metadataPaths...)
+
+	// We should generate new UUID, otherwise /sys/class/dmi/id/product_uuid is identical on all VMs.
```

Comment:
Perhaps mark as a TODO.
I thought "there are loads of ways to generate UUIDs from Go" but then realised UUID generation wasn't the problem, but finding the API to provide it is...
Not something that VF allows you to provide via metadata?

Reply:
Not that I am aware of. The docs are somewhat sparse, to put it mildly, but I couldn't find it anywhere. I will mark as `TODO`